lib/msgpack/core_ext.rb

```diff
@@ -0,0 +1,127 @@
+module MessagePack
+  module CoreExt
+    def to_msgpack(packer_or_io = nil)
+      if packer_or_io
+        if packer_or_io.is_a?(MessagePack::Packer)
+          _to_msgpack packer_or_io
```

Comment:
Maybe name `_to_msgpack` –> `to_msgpack_with_packer`? And perhaps make it private? :bowtie: